content/blog/posts/2022-06-23-welcome-to-shepherd-10

```diff
@@ -0,0 +1,89 @@
+---
+
+## The latest release 🎉
+![shepherd screen shot](../../../static/img/blog/shepherd6-site4.png)
+Following the version 9 release in January 2022, Ship Shape's next major release is here with [Shepherd version 10](https://github.com/shipshapecode/shepherd). This new release delivers more flexible tour options and even speedier run time for highly dynamic content. 
+
+## Where can I attach my Shepherd Step?
+In previous iterations of Shepherd we can define where the `Step` is placed on the DOM with the `attachTo`  object passed as an argument of the options parameter either when setting up a new `Tour` or new `Step` like so:
+
+```js
+// Tour
+const tour = new Tour({
+	steps: [{ attachTo: { element: '.selector', on: 'top' }, ...moreOptions}]
+})
+
+// Step
+const step = new Step(tour, {
+	attachTo: { element: '.selector' , on: 'top'},
+	...moreOptions
+})
+```
+Previously the `element` value could be one of two things, a **DOM selector string** or **HTMLElement**.  This works well for static sites and other cases when the element is loaded at build time and is not expected to change. But what about Single Page Applications ([SPAs](https://developer.mozilla.org/en-US/docs/Glossary/SPA))? What any other situation that requires highly dynamic content?
+
+Shepherd 10 now supports a **callback function** that returns a DOM selector string, HTMLElement, undefined, or null as an `element` value. 
+
+```js
+const step = new Step(tour, {
+	attachTo: { element: () => { '.selector' || HTMLElement || '[data-test-...]' || undefined || null },
+	on: 'top'}
+}) 
+```
+
+We now resolve the `attachTo.element` in the `before-show` phase which by its name resolves the attachTo specifications just once before the Shepherd Step is shown on the DOM. By also **memoizing** the attachTo resolution, the Shepherd 10 delivers a more performant experience to users as they navigate each step in a tour.
+
+## Memo-what?
+
+If you haven't heard of memoization or memoize before, not to worry it is simply a way of storing information which in most cases saves time.
+
+> In programming, **memoization is an optimization technique** that makes applications more efficient and hence faster. It does this by storing computation results in cache, and retrieving that same information from the cache the next time it's needed instead of computing it again. --[Germán Cocca](https://www.freecodecamp.org/news/author/gercocca/)
+
+<iframe src="https://gifer.com/embed/EVRC" width=480 height=276.480 frameBorder="0" allowFullScreen></iframe>
+
+It's not surprising that the words `memorize` and `memoize` share the same latin root. Instead of constantly wondering what is in the fridge, writing down your list and carrying it with you to the grocery store is a much cleaner active recall. For those familair with complexity, memoization is often described as a trade of space complexity for time complexity. We take up a bit more space in the cache with the intention of faster load time.
```

Comment:
I think this is a nice real-world analogy!
Wondering if something like this could make more sense? I looked up "active recall" definition and it's something like "the process of remembering information." I'd think the grocery list helps you _not_ have to remember/recall it since it is the 'memo' already established

Instead of constantly wondering what is or isn't in your fridge, if you write down a list and carry it with you to the grocery store, you only have to think about it once.